lib/__tests__/standalone-globs.test.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+/* eslint-disable node/no-extraneous-require */
+
+const describe = require('@jest/globals').describe;
+const expect = require('@jest/globals').expect;
+const it = require('@jest/globals').it;
+
+/* eslint-enable */
+
+const path = require('path');
+const replaceBackslashes = require('../testUtils/replaceBackslashes');
+const standalone = require('../standalone');
+
+const fixturesPath = replaceBackslashes(path.join(__dirname, 'fixtures', 'globs'));
+
+// Tests for https://github.com/stylelint/stylelint/issues/4521
+
+describe('standalone globbing', () => {
+	describe('paths with special characters', () => {
+		// ref https://github.com/micromatch/micromatch#matching-features
+		const fixtureDirs = [
+			`[digit]/not-digits`,
+			`with spaces`,
+			`extglob!(s)`,
+			`got!negate/negate`,
+			// `extglob+(s)`, // Note: +'s cause errors. Ignoring until it becomes a problem
```

Comment:
I'm curious, what's the error here? Should we document that there's a problem with the `+` character?

Reply:
Ah, I should have made a note of it. The `+` seemed to be escaped differently for this case, causing the expected match to fail. 

I can't remember the specific details, but I think I tracked it down to somewhere in one of globby's dependencies. I thought I'd read an issue that described it, but now I can't find it 🤷 

It seemed like enough of an edge-case that we can skip it for now, and investigate later if it becomes a problem.